### ISSUE TITLE
Notification-api: Upgrade Werkzeug >= 3.x

### ISSUE THREAD

**Issue by edDocMe360** [department-of-veterans-affairs/notification-api#1571]
## User Story - Business Need
There is a dependabot PR (see Potential Dependencies) to update the werkzeug package. There are 8 tests that fail when the package is updated.

- [ ] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)

**As a** notify engineer  
**I want to** update the werkzeurg package
**So that** we have an up-to-date and secure platform.  

### Additional Info and Resources

These are the failing tests:
```
FAILED tests/app/oauth/test_rest.py::TestSsoCommon::test_redirects_to_ui_and_sets_token_in_cookie[authorize] - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
FAILED tests/app/oauth/test_rest.py::TestSsoCommon::test_redirects_to_ui_and_sets_token_in_cookie[callback] - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
FAILED tests/app/oauth/test_rest.py::TestAuthorize::test_should_redirect_to_login_failure_if_organization_membership_verification_or_user_info_retrieval_fails[OAuthException] - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
FAILED tests/app/oauth/test_rest.py::TestAuthorize::test_should_redirect_to_login_failure_if_organization_membership_verification_or_user_info_retrieval_fails[HTTPError] - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
FAILED tests/app/oauth/test_rest.py::TestAuthorize::test_should_redirect_to_login_denied_if_user_denies_access - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
FAILED tests/app/oauth/test_rest.py::TestAuthorizeWhenVaSsoToggleIsOff::test_should_redirect_to_login_failure_if_incorrect_github_id - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
FAILED tests/app/oauth/test_rest.py::TestAuthorizeWhenVaSsoToggleIsOff::test_should_redirect_to_ui_if_user_is_member_of_va_organization - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
FAILED tests/app/oauth/test_rest.py::TestLogout::test_should_redirect_to_ui_and_clear_cookies - AttributeError: 'FlaskClient' object has no attribute 'cookie_jar'
===== 8 failed, 3470 passed, 150 skipped, 35 xfailed in 182.06s (0:03:02) ======

```

- Upgrading Werkzeug>=[3.x](https://github.com/pallets/werkzeug/releases/tag/3.0.0) causes the following unit tests to fail.
- This had breaking changes - 3.x is an upgrade with breaking changes. [release notes](https://werkzeug.palletsprojects.com/en/3.0.x/changes/#version-3-0-0).
- Possible solution [here](https://github.com/miguelgrinberg/Flask-SocketIO/issues/1982).

## Engineering Checklist
- [ ] Reach out to @nathanbwright regarding portal usage of these tests
- [ ] Update the werkzeug package to a version >= 3.0.1 
- [ ] Fix failing unit tests (if applicable)
- [ ] Update werkzeugs package in the API repo so the dependabot issue can be cleared
- [ ] Validate with API deploy of this work

## Acceptance Criteria

- All tests pass in the api repo
- No functionality is affected

## QA Considerations
- Check for any Twistlock issues to clear
- Run regression in Perf

## Potential Dependencies

This is necessary to clear this [dependabot PR](https://github.com/department-of-veterans-affairs/notification-api/pull/1515) for the api repo.

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1571?planning-poker) with Zenhub @cris-oddball @edDocMe360 @EvanParish @k-macmillan @kalbfled @ldraney @nikolai-efimov

**Comment by ldraney** [comment]
Please reach out to Nathan to see if this package is even relevant to current functionality, as this could significantly reduce the work on this ticket -- a thought from Kyle.

**Comment by cris-oddball** [comment]
Closed by #1647
